File: 11-container-with-most-water/11-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int start = 0;
        int end = height.length -1 ;
        int amt = 0;
        while(start < end){
            
            int min_hgt = Math.min(height[start] , height[end]);
            
            amt = Math.max(amt, min_hgt * (end -start));
            
            if(height[start] < height[end]){
                start++;
            }
            
            else{
                end--;
            }
            
            
        }
        
        return amt;
        
    }
}

File: 11-container-with-most-water/11-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        start = 0
        end = len(height) - 1
        
        max_amt = 0
        while start < end:
            min_height = min(height[start],height[end])
            x_distance = end -start
            
            max_amt = max(max_amt,(min_height * x_distance))
            
            if height[start] < height[end]:
                start += 1
            else:
                end -= 1
        return max_amt
            
        
        
        